Tests/MongoSwiftTests/AsyncTestUtils.swift

```diff
@@ -37,6 +37,100 @@ extension MongoSwiftTestCase {
         defer { client.syncCloseOrLogError() }
         return try f(client)
     }
+
+    /// Creates the given namespace using the given client and passes handles to it and its parent database to the given
+    /// function. After the function executes, the collection associated with the namespace is dropped.
+    ///
+    /// Note: If a collection is not specified as part of the input namespace, this function will throw an error.
+    internal func withTestNamespace<T>(
+        client: MongoClient,
+        ns: MongoNamespace? = nil,
+        options: CreateCollectionOptions? = nil,
+        _ f: (MongoDatabase, MongoCollection<Document>) throws -> T
+    ) throws -> T {
+        let ns = ns ?? self.getNamespace()
+
+        guard let collName = ns.collection else {
+            throw TestError(message: "missing collection")
+        }
+
+        let database = client.db(ns.db)
+        let collection: MongoCollection<Document>
+        do {
+            collection = try database.createCollection(collName, options: options).wait()
```

Comment:
I added this new behavior that will drop a collection if it already exists, since I kept having to manually drop stuff left over from failed tests.